src/components/graphs/ScatterGraph.vue

```diff
@@ -0,0 +1,499 @@
+  if (trendLine) {
+    const regression = calculateLinearRegression(sortedData.value);
+    trendLine
+      .style('opacity', 0.7)
+      .transition()
+      .duration(props.animationDuration)
+      .delay(props.animationDuration * 0.2)
+      .ease(d3.easeQuadOut)
+      .attr('x2', xScale(regression.x2))
+      .attr('y2', yScale(regression.y2));
+  }
+
+  hasAnimated.value = true;
+};
+
+const calculateLinearRegression = (data: DataPoint[]): RegressionLine => {
+  const n = data.length;
+  const sumX = data.reduce((sum, d) => sum + d.year, 0);
+  const sumY = data.reduce((sum, d) => sum + d.value, 0);
+  const sumXY = data.reduce((sum, d) => sum + d.year * d.value, 0);
+  const sumXX = data.reduce((sum, d) => sum + d.year * d.year, 0);
+
+  const slope = (n * sumXY - sumX * sumY) / (n * sumXX - sumX * sumX);
+  const intercept = (sumY - slope * sumX) / n;
+
+  const xMin = Math.min(...data.map((d) => d.year));
+  const xMax = Math.max(...data.map((d) => d.year));
+
+  return {
+    x1: xMin,
+    y1: slope * xMin + intercept,
+    x2: xMax,
+    y2: slope * xMax + intercept,
+  };
+};
+
+let intersectionObserver: IntersectionObserver | undefined;
+
+onMounted(() => {
+  // Render chart structure immediately
+  renderChartStructure();
+
+  // Set up intersection observer for data animation
+  intersectionObserver = new IntersectionObserver(
+    (entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !hasAnimated.value &&
+          entry.intersectionRatio >= 0.5 &&
+          chartRendered.value
+        ) {
+          console.log('Animating chart data for:', props.containerId);
+          animateDataElements();
+        }
+      });
+    },
+    {
+      threshold: 0.5,
+    },
+  );
+
+  requestAnimationFrame(() => {
+    if (intersectionObserver && chartContainer.value) {
+      intersectionObserver.observe(chartContainer.value);
+    }
+  });
+});
+
+onBeforeUnmount(() => {
+  if (intersectionObserver && chartContainer.value) {
+    intersectionObserver.unobserve(chartContainer.value);
+    intersectionObserver.disconnect();
+  }
+});
+
+watch(
+  () => props.data,
+  () => {
+    loading.value = true;
+    hasAnimated.value = false;
+    chartRendered.value = false;
+    renderChartStructure();
+    animateDataElements();
+  },
+  { deep: true },
+);
+</script>
+
+<style scoped>
+.scatterplot-container {
+  width: 100%;
+  position: relative;
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto',
+    sans-serif;
+}
+
+@media screen and (min-width: 500px) {
+  .scatterplot-container {
+    max-width: 1000px;
+  }
+}
+
+.loading-overlay {
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background: rgba(255, 255, 255, 0.9);
+  border-radius: 12px;
+  z-index: 1000;
+}
+
+.loading-spinner {
+  width: 40px;
+  height: 40px;
+  border: 4px solid #e2e8f0;
```

Comment:
See other comment about colors, but also we have a `spacing.scss` file you can use for the border value (`$border-v-thick`)